exp/api/v1beta2/rosaroleconfig_types.go

```diff
@@ -0,0 +1,172 @@
+
+// ROSARoleConfigSpec defines the desired state of ROSARoleConfig
+
+type ROSARoleConfigSpec struct {
+	AccountRoleConfig  AccountRoleConfig             `json:"accountRoleConfig"`
+	OperatorRoleConfig OperatorRoleConfig            `json:"operatorRoleConfig"`
+	OIDCConfig         OIDCConfig                    `json:"oidcConfig"`
+	IdentityRef        *infrav1.AWSIdentityReference `json:"identityRef,omitempty"`
+	Region             string                        `json:"region,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=rosaroleconfigs,scope=Namespaced,categories=cluster-api,shortName=rosarole
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+
+// ROSARoleConfig is the Schema for the rosaroleconfigs API
+type ROSARoleConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ROSARoleConfigSpec   `json:"spec,omitempty"`
+	Status ROSARoleConfigStatus `json:"status,omitempty"`
+}
+
+type AccountRoleConfig struct {
+	// TODO: verify max len
+	// +kubebuilder:validation:MaxLength:=4
+	// +kubebuilder:validation:Required
+	Prefix string `json:"prefix"`
+	// +optional
+	PermissionsBoundaryARN string `json:"permissionsBoundaryARN,omitempty"`
+	// +optional
+	Path string `json:"path,omitempty"`
+	// +kubebuilder:validation:Required
+	Version string `json:"version"`
+	// +optional
+	SharedVPCConfig SharedVPCConfig `json:"sharedVPCConfig,omitempty"`
+}
+
+type OperatorRoleConfig struct {
+	// +kubebuilder:validation:MaxLength:=4
+	// +kubebuilder:validation:Required
+	Prefix string `json:"prefix"`
+	// +optional
+	PermissionsBoundaryARN string `json:"permissionsBoundaryARN,omitempty"`
+	// +kubebuilder:validation:Required
+	OIDCConfigID string `json:"oidcConfigId,omitempty"`
+	// +optional
+	SharedVPCConfig SharedVPCConfig `json:"sharedVPCConfig,omitempty"`
+}
+
+type SharedVPCConfig struct {
+	RouteRoleARN       string `json:"routeRoleARN,omitempty"`
+	VPCEndpointRoleARN string `json:"vpcEndpointRoleArn,omitempty"`
+}
+
+type OIDCConfig struct {
+	ManagedOIDC           bool                                      `json:"managedOIDC"`
+	ExternalAuthProviders []rosacontrolplanev1.ExternalAuthProvider `json:"externalAuthProviders,omitempty"`
+}
+
+type Issuer struct {
+	IssuerURL string   `json:"issuerURL"`
+	Audiences []string `json:"audiences,omitempty"`
+}
+
+type ClaimMappings struct {
+	Username Mapping `json:"username"`
+	Groups   Mapping `json:"groups,omitempty"`
+}
+
+type Mapping struct {
+	Claim        string `json:"claim"`
+	PrefixPolicy string `json:"prefixPolicy,omitempty"`
+}
+
+type OIDCClient struct {
+	ComponentName      string          `json:"componentName"`
+	ComponentNamespace string          `json:"componentNamespace"`
+	ClientID           string          `json:"clientID"`
+	ClientSecret       SecretReference `json:"clientSecret"`
+}
```

Comment:
I believe after defining the auteScale struct in the controlPlane no loop dependency to refer to [external_auth_types](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/controlplane/rosa/api/v1beta2/external_auth_types.go)